src/Numerics/SmoothFunctions.cpp

```diff
@@ -0,0 +1,83 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include <stdexcept>
+#include "Numerics/SmoothFunctions.hpp"
+
+namespace qmcplusplus
+{
+
+/** wraps untemplated cmath function
+ *  unless wall clock is actually effected considering dropping cmath dependence 
+ */
+template<typename T>
+void sincosT(T x, T* y, T* z);
```

Comment:
sincos is managed at
config/stdlib/math.h

Reply:
What happens if I want to build both float and double?

Reply:
some platform doesn't have sincosf or a slow sincosf.

Reply:
Actually I don't understand your question. We always build both float and double. The config/stdlib/math.h makes sure sincos for float and double can always be called.

Reply:
Whoa I'd forgotten that file, the headers lead me to believe we were really using cmath sincos.  I don't have a great feelling about this sort of technique.  libc calls should just be the libc calls.

Reply:
That file used to handle more math functions. Now C++11 ships more and we can do less.
Probably it is time to change the name. I just don't like a separate implementation floating.

Reply:
But it does look like we can't quite, I will point at config/stdlib/math.h to avoid the evil of two different overrides.